src/services/PostService.ts

```diff
@@ -25,12 +25,15 @@ export class PostService {
     });
   }
 
-  public async getPostById(params: UuidParam): Promise<PostModel> {
+  public async getPostById(user: UserModel, params: UuidParam): Promise<PostWithIsSaved> {
     return this.transactions.readOnly(async (transactionalEntityManager) => {
       const postRepository = Repositories.post(transactionalEntityManager);
-      const post = await postRepository.getPostById(params.id);
+      let post = await postRepository.getPostById(params.id);
       if (!post) throw new NotFoundError('Post not found!');
-      return post;
+      const userRepository = Repositories.user(transactionalEntityManager);
+      let postWithIsSaved = post as unknown as PostWithIsSaved;
+      postWithIsSaved.isSaved = await userRepository.isSavedPost(user.id, params.id);
```

Comment:
I recommend adding some default logic where `isSaved` returns`false` in case `user` is `undefined`. This would allow functionality for seeing posts without necessarily logging in and handles unexpected calls from the frontend where `user` is `undefined` (I think this might happen if `getPostById` is called without a valid session token)

Reply:
right now if the session token is not valid frontend will get a 403 - user unauthorized error. This comes from using the @currentUser decorator. If we want to unprotect this endpoint we can instead pass in user id and have that be an optional parameter.